### PR TITLE
Dbfs HF

### PR DESCRIPTION
**Pull request by KuuCi**
This PR is to allow DBFS UC_Volume paths to be set in the `hf_name` and properly link the datasource in MLFlow

https://github.com/mosaicml/llm-foundry/pull/1142
Was hitting git issues so making new PR

https://dbc-04ac0685-8857.staging.cloud.databricks.com/ml/experiments/974778159721961/runs/c716ca6ae01045638f4aea44340e2e0b?o=3360802220363900

### CODE REVIEW

File: llmfoundry/utils/config_utils.py [hunk 1]
```diff
@@ -592,7 +592,24 @@ def _process_data_source(
...
+            assert cfg_split
+            from llmfoundry.data.finetuning.tasks import SUPPORTED_EXTENSIONS
+            possible_files = [
+                f'{cfg_split}{ext}' for ext in SUPPORTED_EXTENSIONS
+            ]
+            for file in possible_files:
+                path = os.path.join(hf_path[len('dbfs:'):], file)
+                if _verify_uc_path(path):
+                    data_paths.append(('uc_volume', path, true_split))
+                    unsupported_file = True
```
Comment (dakinggg) [inline llmfoundry/utils/config_utils.py:606]: this should be set to false here, right?

File: llmfoundry/utils/config_utils.py [hunk 2]
```diff
@@ -656,3 +673,38 @@ def _log_dataset_uri(cfg: Dict[str, Any]) -> None:
...
+            'Cannot verify the path of `UCVolumeDatasetSource` due to a connection failure ' + \
+            'with Databricks workspace. Please run `mlflow.login()` to log in to Databricks. ' + \
+            'This does not block creating `UCVolumeDatasetSource`, but your ' + \
+            '`UCVolumeDatasetSource` might be invalid.')
+        return False
+
+    try:
+        w.files.get_metadata(path)
+        return True
+    except Exception:
```
Comment (dakinggg) [inline llmfoundry/utils/config_utils.py:709]: does this need to be generic exception or can it be more specific?

File: llmfoundry/utils/config_utils.py [hunk 3]
```diff
@@ -656,3 +673,38 @@ def _log_dataset_uri(cfg: Dict[str, Any]) -> None:
...
+        w = WorkspaceClient()
+    except ImportError:
+        log.warning(
+            'Cannot verify the path of `UCVolumeDatasetSource` because of missing' + \
+            '`databricks-sdk`. Please install `databricks-sdk` via ' + \
+            '`pip install -U databricks-sdk`. This does not block creating ' + \
+            '`UCVolumeDatasetSource`, but your `UCVolumeDatasetSource` might be invalid.',
+        )
+        return False
+    except Exception:
```
Comment (dakinggg) [inline llmfoundry/utils/config_utils.py:698]: same, does this need to be generic?

File: llmfoundry/utils/config_utils.py (hunk 1 above)
Comment (KuuCi) [reply llmfoundry/utils/config_utils.py:606]: line 606 should be False, if we detect that the file is supported, we should cast a warning

File: llmfoundry/utils/config_utils.py (hunk 3 above)
Comment (KuuCi) [reply llmfoundry/utils/config_utils.py]: Should be value error, here is traceback:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/databricks/sdk/config.py", line 434, in _init_auth
    self._header_factory = self._credentials_provider(self)
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/databricks/sdk/credentials_provider.py", line 615, in __call__
    raise ValueError(
ValueError: cannot configure default credentials, please check https://docs.databricks.com/en/dev-tools/auth.html#databricks-client-unified-authentication to configure credentials for your preferred authentication method.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/databricks/sdk/config.py", line 100, in __init__
    self._init_auth()
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/databricks/sdk/config.py", line 439, in _init_auth
    raise ValueError(f'{self._credentials_provider.auth_type()} auth: {e}') from e
ValueError: default auth: cannot configure default credentials, please check https://docs.databricks.com/en/dev-tools/auth.html#databricks-client-unified-authentication to configure credentials for your preferred authentication method.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/v.chen/Desktop/Experiments/debug_oncall.py", line 31, in <module>
    print(_verify_uc_path('ooga'))
          ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/v.chen/Desktop/Experiments/debug_oncall.py", line 15, in _verify_uc_path
    w = WorkspaceClient()
        ^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/databricks/sdk/__init__.py", line 121, in __init__
    config = client.Config(host=host,
             ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/databricks/sdk/config.py", line 105, in __init__
    raise ValueError(message) from e
ValueError: default auth: cannot configure default credentials, please check https://docs.databricks.com/en/dev-tools/auth.html#databricks-client-unified-authentication to configure credentials for your preferred authentication method.

File: llmfoundry/utils/config_utils.py (hunk 2 above)
Comment (KuuCi) [reply llmfoundry/utils/config_utils.py]: should be Attribute Error, here is traceback:

v.chen@D72F32QJ3J Experiments % python3 debug_oncall.py
Traceback (most recent call last):
  File "/Users/v.chen/Desktop/Experiments/debug_oncall.py", line 28, in <module>
    print(_verify_uc_path('ooga'))
          ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/v.chen/Desktop/Experiments/debug_oncall.py", line 25, in _verify_uc_path
    w.files.get_metadata(path)
    ^^^^^^^^^^^^^^^^^^^^
AttributeError: 'FilesAPI' object has no attribute 'get_metadata'

File: llmfoundry/utils/config_utils.py [hunk 4]
```diff
@@ -592,7 +592,24 @@ def _process_data_source(
     elif 'hf_name' in dataset:
         hf_path = dataset['hf_name']
         backend, _, _ = parse_uri(hf_path)
-        if backend:
+        unsupported_file = True
+        if hf_path.startswith('dbfs:'):
```
Comment (dakinggg) [inline llmfoundry/utils/config_utils.py:596]:
```suggestion
        if backend == 'dbfs':
```

File: llmfoundry/utils/config_utils.py [hunk 5]
```diff
@@ -592,7 +592,24 @@ def _process_data_source(
...
+            assert cfg_split
+            from llmfoundry.data.finetuning.tasks import SUPPORTED_EXTENSIONS
+            possible_files = [
+                f'{cfg_split}{ext}' for ext in SUPPORTED_EXTENSIONS
+            ]
+            for file in possible_files:
+                path = os.path.join(hf_path[len('dbfs:'):], file)
```
Comment (dakinggg) [inline llmfoundry/utils/config_utils.py:603]: we can get path from the output of `parse_uri` right?

File: llmfoundry/utils/config_utils.py (hunk 2 above)
Comment (dakinggg) [reply llmfoundry/utils/config_utils.py]: the error you put is not an attribute error right? its a not found error?

**Comment by KuuCi** [comment]
https://dbc-04ac0685-8857.staging.cloud.databricks.com/ml/experiments/974778159721961/runs/910967976f4d4914b0422e38b587e9b3?o=3360802220363900

**Comment by KuuCi** [comment]
https://dbc-04ac0685-8857.staging.cloud.databricks.com/ml/experiments/974778159721961/runs/36fb3460756e46b6a4c59923b170e7e1?o=3360802220363900

File: llmfoundry/utils/config_utils.py [hunk 6]
```diff
@@ -656,3 +676,39 @@ def _log_dataset_uri(cfg: Dict[str, Any]) -> None:
...
+    except ImportError:
+        log.warning(
+            'Cannot verify the path of `UCVolumeDatasetSource` because of missing' + \
+            '`databricks-sdk`. Please install `databricks-sdk` via ' + \
+            '`pip install -U databricks-sdk`. This does not block creating ' + \
+            '`UCVolumeDatasetSource`, but your `UCVolumeDatasetSource` might be invalid.',
+        )
+        return False
+    except Exception:
+        log.warning(
```
Comment (dakinggg) [inline llmfoundry/utils/config_utils.py:703]: error message should not be specific if error is generic, and message should include the actual errro

**Review by dakinggg — APPROVED** [review]

**Comment by irenedea** [comment]
@KuuCi could you add a more descriptive PR title?

**Pull request merged**